include/pybind11/attr.h

```diff
@@ -545,6 +549,7 @@ template <typename... Extra,
           size_t named = constexpr_sum(std::is_base_of<arg, Extra>::value...),
           size_t self  = constexpr_sum(std::is_same<is_method, Extra>::value...)>
 constexpr bool expected_num_args(size_t nargs, bool has_args, bool has_kwargs) {
+    PYBIND11_WORKAROUND_INCORRECT_MSVC_C4100(nargs, has_args, has_kwargs);
```

Comment:
This is a really buggy warning, I guess? Because the parameters are literally used on the next line.

Reply:
> This is a really buggy warning, I guess?

Absolutely. We're working around a bug in old MSVC versions.